tests/Test/Foundation/String.hs

```diff
@@ -59,6 +59,14 @@ testStringCases =
               "this is only a simple string but quite longer than the 64 bytes used in the modified UTF8 parser"
               "this is only a simple string but quite longer than the 64 bytes used in the modified UTF8 parser"
         ]
+    , Group "CaseMapping" 
+         [ CheckPlan "a should capitalize to A" $ validate "a" $ upper "a" == "A"
+         , CheckPlan "b should capitalize to B" $ validate "b" $ upper "b" == "B"
+         , CheckPlan "B should capitalize to B" $ validate "B" $ upper "B" == "B"
+         , CheckPlan "é should capitalize to É" $ validate "é" $ upper "é" == "É"
+         , CheckPlan "ß should capitalize to SS" $ validate "ß" $ upper "ß" == "SS"
+         , CheckPlan "ﬄ should capitalize to FFL" $ validate "ﬄ" $ upper "ﬄ" == "FFL"
```

Comment:
Can you please add a random test that capitalising is idempotent or something? Anything that hammers on long strings will show up GC and segfault issues which are the most likely problems.

Reply:
Given all the sentinels around 0 please make sure you test both `\0a` and `\0\0` (one that requires transforming and one that does not). I suspect they are broken.

Reply:
Great idea!

Reply:
Alright, I added the property test.

I am a bit confused by your second comment thought. Isn't `\x0a` LF and \0\0 two nulls char? I am assuming you would like a test case where the heading bytes are 0s, would `upper à == À` (`upper \x00e0 == \x00c0`)? 

I am maybe missing the point here...

Reply:
I meant `['\0','a']`. Specifically, try a test where nothing in the string converts (so it fails out early), and a test where they do convert (so you test the conversion path too).

Reply:
Right, got it!

Done.